modules/instance_template/main.tf

```diff
@@ -43,13 +43,18 @@ locals {
   # initialize the block only if it is enabled.
   shielded_vm_configs = var.enable_shielded_vm ? [true] : []
 
-  gpu_enabled            = var.gpu != null
-  alias_ip_range_enabled = var.alias_ip_range != null
+  gpu_enabled                    = var.gpu != null
+  alias_ip_range_enabled         = var.alias_ip_range != null
+  snp_confidential_instance_type = var.confidential_instance_type == "SEV_SNP"
   on_host_maintenance = (
-    var.preemptible || var.enable_confidential_vm || local.gpu_enabled || var.spot
+    var.preemptible || var.enable_confidential_vm || local.gpu_enabled || var.spot || local.snp_confidential_instance_type
```

Comment:
Confidential VM's now support Live migration, so doesn't need to terminate anymore.  however there are some limitations that may warrant additional logic.

(see https://cloud.google.com/confidential-computing/confidential-vm/docs/troubleshoot-live-migration)